Support getter defs in CommonJS exports

kommen opened the pull request:
This fixes #2510

ChadKillingsworth commented on src/com/google/javascript/jscomp/ProcessCommonJSModules.java:
```diff
@@ -878,7 +878,7 @@ private void expandObjectLitAssignment(NodeTraversal t, Node export) {
             value = IR.name(key.getString());
           }
-        } else if (key.isMemberFunctionDef()) {
+        } else if (key.isMemberFunctionDef() || key.isGetterDef()) {
```
This blocks the crash, but doesn't produce correct output.  Either of the following would be acceptable transformations:

```js
/** @const */ var module$test = {
  get a() { return 1; }
};
```

or

```js
Object.defineProperties(module$test, {
  a: {
    get: function() { return 1; };
  }
});
```

brad4d commented:
FYI, @kyshim is working on a simple 'report error instead of crash' fix for #2510 
We don't have time to do more than that right now.

@kommen we'll try out your fix after you address the problem @ChadKillingsworth pointed out.

kommen commented:
All, thanks for the valuable and timely feedback. Much appreciated.

@ChadKillingsworth I've addressed your comments.
I'm still trying to make consuming `htmlparser2` (see #2510)  working. In my tests, the compiler now no longer crashes, but I get multiple warnings `JSC_COMMONJS_SUSPICIOUS_EXPORTS_ASSIGNMENT` and `JSC_JS_MODULE_LOAD_WARNING`

I'm not sure yet if this are different issues or if this PR still doesn't produce invalid output.

brad4d commented:
@kommen 
I took a look at the `htmlparser2` link you had on #2510 
https://github.com/fb55/htmlparser2/blob/2e6c0e0f3f79cc268efc59e73b25d917c07d8493/lib/index.js#L15

I don't think we support using `require()` inside an exported object literal as is done there.
@ChadKillingsworth 
am I wrong about that?

If I'm not wrong, that's possibly a bigger problem that needs solving.

ChadKillingsworth commented:
@brad4d I actually don't know. I'll run some test cases to see.

kommen commented:
I rebased & squashed my commits onto current master, if you're still interested in merging this.
